### ISSUE TITLE
Better Heuristic For Finding Openflow Executable

### ISSUE THREAD

**Issue by craig-riecke** [frenetic-lang/frenetic#508]
When Frenetic runs with http-controller, it starts up the separate executable `openflow.native` as a subprocess.  This works fine for programming in Python or REST.   However, it doesn't work well when building a monolithic OCaml Frenetic app (basically the Northbound app is compiled in with Frenetic itself).  

Currently the workaround is to create a symbolic link to `openflow.native` in the same directory as the monolithic app executable, as show in the Wiki.  It'd be nice not to require that.

**Comment by smolkaj** [comment]
Just compile with `make install`. Then `openflow` should be on the PATH.

**Comment by craig-riecke** [comment]
Yeah, that would work ... if a monolithic app looked on the PATH.  It doesn't.  It only looks for `openflow` or `openflow.native` in the same directory as the executable you're running.   The logic is in Frenetic_OpenFlow0x01_Controller.ml line 30.  

It should be an easy fix.

**Comment by jnfoster** [comment]
Wait. Why do we have two executables anyway? It seems like it only causes
confusion with PATH, etc. I think we should run frenetic.native twice (or n
times) with different command-line arguments -- once for the high-level
controller and once for the low-level controller. The first one could even
fork() to create the other copies so we only need one invocation at the
command line.

-N

**Comment by craig-riecke** [comment]
@jnfoster We do use one invocation - the frenetic.native binary (or the monolithic app) starts the openflow.native binary through Process.create.  It would be great to use fork instead, but we can't  because of Async.  Or rather, the Async scheduler - we start it up to handle some deferred's and ivar's (they are needed for other parts of Frenetic) prior to creating the openflow.native process.  If we forked, the Async scheduler would be totally mucked up in the child process, ref: https://groups.google.com/forum/#!topic/ocaml-core/an6dwjN_yCY

We could do one binary, but a monolithic app would have the same problem finding frenetic.native - Process.create doesn't look through the PATH by default.  Unless maybe it started another copy of itself...

**Comment by smolkaj** [comment]
good point...

**Comment by jnfoster** [comment]
Not sure which point is a good one.

It would still be my preference to have a single binary -- it eliminates
complexity in several places.

-N

**Comment by smolkaj** [comment]
"Good point" was referring to Nate's comment. Having two binaries trying to find each other sounds like it will break very easily. There must be a way around it.

Craig, under the link you posted they mention [`reset_in_forked_process`](https://ocaml.janestreet.com/ocaml-core/113.33/doc/async/Async/Std.mod/Scheduler.mod/#/reset_in_forked_process.val). Can't we use that?
